torchmetrics/metric.py

```diff
@@ -256,7 +257,10 @@ def reset(self):
         """
         This method automatically resets the metric state variables to their default value.
         """
-        self._computed = None
+        # lower lightning versions requires this implicitly to log metric objects correctly
+        # in self.log
+        if not _LIGHTNING_AVAILABLE or _LIGHTNING_GREATER_EQUAL_1_3:
+            self._computed = None
```

Comment:
isn't this backwards? shouldn't lightning be accounting for the torchmetrics version, not the other way around?

Reply:
since we cannot change earlier versions of lightning, we need to do the change in torchmetrics